reactive/beats_base.py

```diff
@@ -27,11 +27,17 @@ def cache_logstash_data(logstash):
         hosts = cache.get('beat.logstash')
     else:
         hosts = []
+    removed_hosts = hosts.copy()
     for unit in units:
         host_string = "{0}:{1}".format(unit['private_address'],
                                        unit['port'])
         if host_string not in hosts:
             hosts.append(host_string)
+        else:
+            removed_hosts.remove(host_string)
+    if len(removed_hosts) != 0:
+        for host in removed_hosts:
+            hosts.remove(host)
```

Comment:
Any trouble with this.  In what situation would `len(removed_hosts) == 0)` be `True` but the following would fail?
```suggestion
    for host in removed_hosts:
        hosts.remove(host)
```

Reply:
I don't think there is a case that the modified code would fail. However, I just re-analyzed the code I added, it's effect is actually equivalent to simply initializing `hosts` as an empty list (as in `cache_elasticsearch_data` and `cache_kafka_data`). Hence I will spend sometime refactoring the code entirely and request another review afterwards. Thanks again :)

Reply:
awesome.  yeah, i don't know why the `logstash` handler is really any different from the others.  thanks!